common/src/main/java/tds/common/logging/EventLogger.java

```diff
@@ -0,0 +1,175 @@
+        return extraFields.get(key);
+    }
+
+    /**
+     * Log event occurrences to centralized logging.
+     * <p>
+     * These events are intended to be sent to a logstash server and searchable using elastic search.
+     *
+     * @param app        name of the app writing the log
+     * @param logEvent   name of the event occurring
+     * @param checkpoint an optional checkpoint name
+     * @param message    an optional free-format message string to add to the log entry
+     * @param data       optional searchable event data to add to the recorded event
+     */
+    public void info(final String app, final String logEvent, final String checkpoint,
+                     final String message, final Map<EventData, Object> data) {
+        log(formatMessage(logEvent, message), getFieldMap(app, checkpoint, data));
+    }
+
+    /**
+     * Log event errors to centralized logging.
+     * <p>
+     * These events are intended to be sent to a logstash server and searchable using elastic search.
+     *
+     * @param app        name of the app writing the log
+     * @param logEvent   name of the event occurring
+     * @param checkpoint an optional checkpoint name
+     * @param message    an optional free-format message string to add to the log entry
+     * @param data       optional searchable event data to add to the recorded event
+     */
+    public void error(final String app, final String logEvent, final String checkpoint,
+                      String message, final Map<EventData, Object> data, final Exception e) {
+        error(formatMessage(logEvent, message), getFieldMap(app, checkpoint, data), e);
+    }
+
+    public void setBeginMillis(long beginMillis) {
+        this.beginMillis = beginMillis;
+    }
+
+    public void setEndMillis(long endMillis) {
+        this.endMillis = endMillis;
+    }
+
+    private void log(final String event, final Map<String, Object> fields) {
+        try {
+            logger.info(marker, format("EVENT:%s JSON:({\"event_data\": %s})", event, writer.writeValueAsString(fields)));
+        } catch (Exception e) {
+            logger.warn(marker, format("exception occurred while logging event: %s", event), e);
+        }
+    }
+
+    private void error(final String event, final Map<String, Object> fields, final Exception e) {
+        try {
+            logger.error(marker, format("EVENT:%s JSON:({\"event_data\": %s})", event, writer.writeValueAsString(fields)),
+                e);
+        } catch (Exception inner) {
+            logger.error(marker, format("exception occurred while error event: %s", event), inner);
+        }
+    }
+
+    private void addTimerMetrics(Map<EventData, Object> fields) {
+        fields.put(EventData.MILLIS_CURRENT, System.currentTimeMillis());
+        if(beginMillis != 0) {
+            fields.put(EventData.MILLIS_BEGIN, beginMillis);
+            fields.put(EventData.ELAPSED_TIME, System.currentTimeMillis() - beginMillis);
```

Comment:
Should the ELAPSED_TIME instead be moved down into the if(endMillis != 0) block and be based off of the MILLIS_END property?